console/module/device/src/main/java/org/eclipse/kapua/app/console/module/device/server/GwtDeviceServiceImpl.java

```
@@ -453,13 +454,21 @@ public void addDeviceTag(GwtXSRFToken xsrfToken, String scopeIdString, String de
             Device device = drs.find(scopeId, deviceId);
 
             Set<KapuaId> tagIds = device.getTagIds();
```

Please throw an exception when the same tag was already added.